src/caches/beliefpropagationcache.jl

```diff
@@ -0,0 +1,221 @@
+default_mt_constructor(inds_e) = ITensor[denseblocks(delta(inds_e))]
+default_mt_storage(ptn::PartitionedGraph) = Dict()
+function default_contractor(contract_list::Vector{ITensor}; kwargs...)
+  return contract_exact(contract_list; kwargs...)
+end
+default_contractor_kwargs() = (; normalize=true, contraction_sequence_alg="optimal")
+@traitfn default_bp_niters(g::::(!IsDirected)) = is_tree(g) ? 1 : nothing
```

Comment:
I would suggest changing all uses of `niters` to `maxiter`, since it is really a maximum number of iterations, since it can be stopped short by the `target_precision`.

`maxiter` is used by both `KrylovKit.jl` (https://jutho.github.io/KrylovKit.jl/stable/man/eig/#KrylovKit.eigsolve) and by `IterativeSolvers.jl` (https://iterativesolvers.julialinearalgebra.org/dev/linear_systems/cg/#CG).

Additionally, I would vote for changing `target_precision` to `tol`, which is also a common name for that kind of keyword argument (used in `KrylovKit.jl`, `IterativeSolvers.jl`, and elsewhere).